packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -242,34 +239,29 @@ export const TreeItem = React.forwardRef(function TreeItem(
 
   React.useEffect(() => {
     // On the first render a node's index will be -1. We want to wait for the real index.
-    if (registerNode && unregisterNode && index !== -1) {
-      registerNode({
+    if (instance && index !== -1) {
```

Comment:
The current `TreeItem` handles when there is no `TreeView` above in the tree (see `const expanded = isExpanded ? isExpanded(nodeId) : false;`).

To avoid any breaking change I added those `instance` check, but for me it make no sense and we could just throw an error in `useTreeViewContext` like we do on `useGridApiContext`.